src/openms/source/CONCEPT/QCMBRalignment.cpp

```diff
@@ -11,90 +11,165 @@ QCMBRalignment::~QCMBRalignment()
-    					//cout << "corrected_RT: " << p_it->getRT()  << endl;
-    					//cout << "original_RT: " << p_it->getMetaValue("original_RT") << endl;
+ 		  else 
+ 		  {
+ 		    // pep_id and sequence + spectrum ident
+ 			for (vector<PeptideIdentification>::iterator p_it = pep_id.begin(); p_it!=pep_id.end();p_it++) 
+ 			{
     					
-    					vector<PeptideHit> hit = p_it->getHits();
-    					AASequence seq = hit.begin()->getSequence();
-    					float corrRT = p_it->getRT();
-    					float orRT = p_it->getMetaValue("original_RT");
-    					String spec = p_it->getMetaValue("spectrum_reference");  //"spectrum=3462"
-    					String spectrum_ref = spec.substr(9);
-    					//cout << spectrum_ref << endl;
-    					//String spectrumID = p_it->getIdentifier();
+    		  vector<PeptideHit> hit = p_it->getHits();
+    		  AASequence seq = hit.begin()->getSequence();
+    		  float corrRT = p_it->getRT();
+    		  float orRT = p_it->getMetaValue("original_RT");
+    		  String spec = p_it->getMetaValue("spectrum_reference");  //"spectrum=3462"
+    		  String spectrum_ref = spec.substr(9);
     					
-    					rawFiles.push_back (rfile);
-    					cout << rfile << endl;
- 						correctedRTs.push_back (corrRT);
- 						cout << corrRT << endl;
- 						originalRTs.push_back (orRT);
- 						cout << orRT << endl;
- 						sequences.push_back (seq.toString());
- 						cout << seq << endl;
- 						spectra.push_back (spectrum_ref);
- 						cout << spec << endl;
     					
-						//for (vector<PeptideHit>::const_iterator h_it = hit.begin(); h_it!=hit.end();h_it++) {
-			    		//	cout << h_it->getSequence() << endl;
-    					//}
-    				}
- 			    }
+    		  rawFiles.push_back (rfile);
+    					//cout << rfile << endl;
+ 			  correctedRTs.push_back (corrRT);
+ 						//cout << corrRT << endl;
+ 			  originalRTs.push_back (orRT);
+ 						//cout << orRT << endl;
+ 			  sequences.push_back (seq.toString());
+ 						//cout << seq << endl;
+ 			  spectra.push_back (spectrum_ref);
+ 						
+    		}
+ 		 }  
  				
     		
- 			}
+ 	  }  
  			
- 	  }
- 	  //MetricMap& raws, MetricMap& outSeq, MetricMap& outCorrRT, MetricMap& outOrigRT, MetricMap& outSpectra
- 	  raws.pushDataString("raw_file",rawFiles);
- 	  outSeq.pushDataString("sequences",sequences);
- 	  outCorrRT.pushDataFloat("corrected_RT",correctedRTs);
- 	  outOrigRT.pushDataFloat("original_RT",originalRTs);
- 	  outSpectra.pushDataString("spectre_id",spectra);
- 	  
- 	  return raws.isEmpty()&&outSeq.isEmpty()&&outCorrRT.isEmpty()&&outOrigRT.isEmpty()&&outSpectra.isEmpty()?0:1;
+ 	}
+ 	  //vector<String> rawFiles;
+  	  //vector<String> sequences;
+  	  //vector<float> originalRTs;
+  	  //vector<float> correctedRTs;
+  	  //vector<String> spectra;
  	  
+ 	//int a = rawFiles.isEmpty()&&sequences.isEmpty()&&originalRTs.isEmpty()&&correctedRTs.isEmpty()&&psectra.isEmpty()?1:0;
+ 	
+ 	 int numOfcorrRT = originalRTs.size();
+        
+     if(numOfcorrRT>0)
```

Comment:
why not fill the mzTab object above, right when the data is generated? This would probably cut the code size in half and is better to read.